Code/Framework/AzFramework/Tests/InputTests.cpp

```diff
@@ -33,7 +34,8 @@ namespace InputUnitTests
         InputTest() : LeakDetectionFixture()
         {
             // Many input tests are only valid if the GamePad device is supported on this platform.
-            m_gamepadSupported = InputDeviceGamepad::GetMaxSupportedGamepads() > 0;
+            auto deviceGamepadImplFactory = AZ::Interface<InputDeviceGamepad::ImplementationFactory>::Get();
+            m_gamepadSupported = (deviceGamepadImplFactory != nullptr) ? (deviceGamepadImplFactory->GetMaxSupportedGamepads() > 0) : false;
```

Comment:
The second of the @lsemp3d pet-peeve special.
The ternary statement where a boolean is assigned, never needs to have a hard code `false` or `true` keyword.
Using logical-AND and logical_OR can always replace the ternary when the value is a boolean.
```suggestion
            m_gamepadSupported = (deviceGamepadImplFactory != nullptr) && (deviceGamepadImplFactory->GetMaxSupportedGamepads() > 0);
```